eng/scripts/TypeSpec-Requirement.ps1

```diff
@@ -0,0 +1,75 @@
+[CmdletBinding()]
+param (
+)
+Set-StrictMode -Version 3
+
+. $PSScriptRoot/ChangedFiles-Functions.ps1
+. $PSScriptRoot/Logging-Functions.ps1
+
+$repoPath = Resolve-Path "$PSScriptRoot/../.."
+$pathsWithErrors = @()
+
+$filesToCheck = @(Get-ChangedSwaggerFiles)
+if (!$filesToCheck) {
+  LogInfo "No swagger files found to check"
+}
+else {
+  # Example: specification/foo/resource-manager/Microsoft.Foo/stable/2023-01-01/Foo.json
+  # - Forward slashes on both Linux and Windows
+  foreach ($file in $filesToCheck) {
+    LogInfo "Checking $file"
+
+    $jsonContent = Get-Content (Join-Path $repoPath $file) | ConvertFrom-Json -AsHashtable
+
+    if ($null -ne ${jsonContent}?["info"]?["x-typespec-generated"]) {
```

Comment:
Just as a FYI for any reader: this needs to be kept in sync with openapi-alps:
https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps?path=/private/azure-swagger-validation/azureSwaggerValidation/src/typespecValidation.ts&version=GBmain&line=214&lineEnd=214&lineStartColumn=62&lineEndColumn=82&lineStyle=plain&_a=contents:

``` typescript
async function getSwaggersGeneratedFromTypeSpec(context:IValidatorContext) {
  const swaagersDiff = await context.getSwaggerDiffs()
  const relatedSwaggers = [...swaagersDiff.additions,...swaagersDiff.changes]
  const result = relatedSwaggers.filter(f => {
    return !!JSON.parse(readFileSync(f).toString())?.info?.["x-typespec-generated"]
  })
  return result
}
```